lib/rspec_api_documentation/dsl/endpoint/params.rb

```diff
@@ -30,6 +30,16 @@ def extended
             unless p[:value]
               cur = extra_params
               [*p[:scope]].each { |scope| cur = cur && (cur[scope.to_sym] || cur[scope.to_s]) }
+
+              # When the current parameter is an array of objects, we use the
+              # first one for the value and add a scope indicator. The
+              # resulting parameter name looks like +props[pictures][][id]+
+              # this.
+              if cur.is_a?(Array) && cur.first.is_a?(Hash)
+                cur = cur.first
+                param[:scope] << ''
```

Comment:
The problem anyone's having is here. 
This line changes `params`, so next time we use `example { do_request }`, it is called with altered `scope`. Causing line 32 to fail.
If we move these lines into a loop, we'll go deeper and deeper, ending up with as many `[]`-s as we had `do_request`-s. E.g. `props[pictures][][][id]` if we called `do_request` inside a group for request twice

I would suggest changing the code in such way that if we want to do nested arrays in `parameter`, we should use `''` there. Like so:
```
parameter :pictures, type: :array
parameter :id, scope: [:pictures, '']
```

I can do it, just need some kind of approval on it =)

Auto-adding `[]` for array scopes of course would be nice, but I can't see a good way of doing it right now (if you do, you are very welcome). I think it should be done somewhere in writers, though, not changing user-defined params.

Or we can just use plain old `scope: 'pictures[]'`